generate.sh

```diff
@@ -68,12 +58,78 @@ while [[ $# > 0 ]]; do
       configuration=$2
       shift
       ;;
+    -maccatalyst)
+      is_maccatalyst=true
+      ;;
+    -tool)
+      tool=true
+      ;;
+    -experimental)
+      experimental=true
+      ;;
+    -help|-h)
+      usage
+      exit 0
+      ;;
   esac
 
   shift
 done
 
+if [[ $platform != "iPhoneOS" && $platform != "iPhoneSimulator" && $platform != "AppleTVOS" && $platform != "AppleTVSimulator" && $platform != "MacOSX" ]]; then
+    echo "Error: Invalid platform '$platform'."
+    usage
+    exit 1
+fi
+
+if [[ -z $version ]]; then
+    version=$(xcrun --sdk "$(echo "$platform" | tr '[:upper:]' '[:lower:]')" --show-sdk-version)
+fi
+
+case "$platform" in
+    "iPhoneOS")
+        target="apple-ios"
+        if $is_maccatalyst; then
```

Comment:
shouldn't `$is_maccatalyst` be treated together with `macOS` rather than `iOS`?

Reply:
Target is `apple-ios-macabi` and SDK is macOS. I made separate case for it.